### PR TITLE
scripts: fix client and tools cover profile

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Ref #4399

### What is changed and how does it work?

- add client cover profile
- make client test with failpoint

```commit-message
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/Makefile [hunk 1]
```diff
@@ -33,9 +33,8 @@ tidy:
 	git diff go.mod go.sum | cat
 	git diff --quiet go.mod go.sum
 
-ci-test-job: failpoint-enable
-	CGO_ENABLED=1 go test ./... -v -tags deadlock -race -cover || { $(MAKE) failpoint-disable && exit 1; }
-	$(MAKE) failpoint-disable
+ci-test-job:
```
Comment (HuSharp) [inline tools/Makefile:36]: There is no need to enable failpoint in this command, `ci-test-job` will turn on failpoint at the beginning. 
disabling useless commands here will reduce some of ci time

**Comment by HuSharp** [comment]
image
can increase 1.23%

File: tools/Makefile (hunk 1 above)
Comment (rleungx) [reply tools/Makefile:36]: Before we can run `make ci-test-job` in client package independently.

Comment (HuSharp) [reply tools/Makefile:36]: Maybe `make test` is better suited for local testing 🤔(although the tools package does not have `make test` now, other Makefiles all have `make test`)  
Or do you think `make test` needs to be merged with `make ci-test-job` as the same command?

Comment (HuSharp) [reply tools/Makefile:40]: support `make test` and `make ci-test-job` currently

**Review by nolouch — COMMENTED** [review]
rest lgtm

File: client/Makefile [hunk 2]
```diff
@@ -12,20 +12,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+ROOT_PATH := ..
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply client/Makefile:15]: Do you mean `ROOT_PATH ?= ..`?

[the remark this answers is not in the document]
Comment (HuSharp) [reply client/Makefile]: addressed

**Review by nolouch — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**